### PR TITLE
Issue/1114 woo product store

### LINKED ISSUE #1114: WooCommerce: Add product store
We're expecting to add product detail to the WC app soon, which will require us to fetch and store product information. At this stage we're simply showing a single product so the associated PR doesn't need to fetch a list of products.

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #1114 - adds a Woo product store & product table along with the ability to fetch and store a single product from the API.

To test, tap the "Fetch single product" button in the example app and enter a remote product ID (you can enter 1537 if you're using Julia's test site). There are also mocked and release tests in this PR.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java [hunk 1]
```diff
@@ -530,6 +534,29 @@ private void migrateAddOn(@AddOn String addOnName, SQLiteDatabase db, int oldDbV
                     db.execSQL("ALTER TABLE WCOrderStatsModel ADD STARTDATE TEXT");
                     db.execSQL("ALTER TABLE WCOrderStatsModel ADD QUANTITY TEXT");
                     break;
+                case 55:
+                    AppLog.d(T.DB, "Migrating addon " + addOnName + " to version " + (oldDbVersion + 1));
+                    db.execSQL("DROP TABLE IF EXISTS WCProductModel");
+                    db.execSQL("CREATE TABLE WCProductModel (_id INTEGER PRIMARY KEY AUTOINCREMENT,"
+                               + "LOCAL_SITE_ID INTEGER,NAME TEXT NOT NULL,SLUG TEXT NOT NULL,PERMALINK TEXT NOT NULL,"
```
Comment (AmandaRiu) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: `REMOTE_PRODUCT_ID` is missing

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java [hunk 2]
```diff
@@ -530,6 +534,29 @@ private void migrateAddOn(@AddOn String addOnName, SQLiteDatabase db, int oldDbV
...
+                               + "DATE_CREATED TEXT NOT NULL,DATE_MODIFIED TEXT NOT NULL,"
+                               + "TYPE TEXT NOT NULL,STATUS TEXT NOT NULL,FEATURED INTEGER,"
+                               + "CATALOG_VISIBILITY TEXT NOT NULL,DESCRIPTION TEXT NOT NULL,"
+                               + "SHORT_DESCRIPTION TEXT NOT NULL,SKU TEXT NOT NULL,"
+                               + "PRICE TEXT NOT NULL,REGULAR_PRICE TEXT NOT NULL, ALE_PRICE TEXT NOT NULL,"
```
Comment (AmandaRiu) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: `ALE_PRICE` should be `SALE_PRICE `

File: example/src/androidTest/java/org/wordpress/android/fluxc/mocked/MockedStack_WCProductsTest.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        val payload = lastAction!!.payload as RemoteProductPayload
+        with(payload) {
+            assertNull(error)
+            assertEquals(remoteProductId, product.remoteProductId)
+            assertEquals(product.getCategories().size, 2)
+            assertEquals(product.getImages().size, 2)
+            assertEquals(product.getTags().size, 2)
+        }
+    }
+
```
Comment (AmandaRiu) [inline example/src/androidTest/java/org/wordpress/android/fluxc/mocked/MockedStack_WCProductsTest.kt]: Recommend adding a method to test the deserialization of the api response to the `WCProductModel`, just to make sure we are good there as well.

**Comment by nbradbury** [comment]
Thanks for the review @AmandaRiu! I fixed those mistakes in the product table definition and added SQL test, so ready for another round.

**Pull request merged**